test/validation/parameterValidation.test.js

```diff
@@ -0,0 +1,292 @@
+            'in': 'path',
+            'type': Joi.number(),
+            'name': 'id',
+            'required': true
+          },
+          {
+            query: {}
+          });
+
+      expect(result)
+        .to.be.an('Object');
+
+      expect(result.id)
+        .to.be.a('string');
+    });
+
+    it('with missing required field', () => {
+      const result =
+        parameterValidation({
+            'in': 'path',
+            'type': Joi.number(),
+            'name': 'id',
+            'required': true
+          },
+          {
+           params: {}
+          });
+
+      expect(result)
+        .to.be.an('Object');
+
+      expect(result.id)
+        .to.be.a('string');
+    });
+
+    it('with incorrect field type', () => {
+      const result =
+        parameterValidation({
+          'in': 'path',
+          'type': Joi.object(),
+          'name': 'id',
+          'required': true
+        });
+
+      expect(result)
+        .to.be.an('Object');
+
+      expect(result.id)
+        .to.be.a('string');
+    });
+  });
+
+  describe('errors', () => {
+    const req = {
+     params: {
+        id: 42
+      }
+    };
+
+    it('with missing schema props', () => {
+      expect(() => {
+        parameterValidation({
+          'in': 'path',
+          'type': Joi.number()
+        }, req);
+      }).to.throw(TypeError);
+    });
+
+    it('with incorrect schema props', () => {
+      expect(() => {
+        parameterValidation({
+          'in': 'path',
+          'type': Joi.number(),
+          'name': () => { },
+          'required': true
+        }, req);
+      }).to.throw(TypeError);
+    });
+
+    it('with no schema', () => {
+      expect(() => {
+        parameterValidation(null, req);
+      }).to.throw(TypeError);
+    });
+  });
+
+  describe('converts', () => {
+    const schema = {
+      'in': 'path',
+      'name': 'id',
+      'required': true
+    };
+
+    it('to a number', () => {
+      schema.type = Joi.number();
+      const req = {
+        params: {
+          id: '1234'
+        }
+      };
+      const validation = parameterValidation(
+        schema,
+        req
+      );
+
+      expect(validation)
+        .to.be.a('boolean');
+
+      expect(validation)
+        .to.be.true;
+
+      expect(req.params.id)
+        .to.be.a('number');
+
+      expect(req.params.id)
+        .to.equal(1234);
+    });
+
+    it('to a date', () => {
+      schema.type = Joi.date();
+      const req = {
+        params: {
+          id: '1943-04-16 00:00:000'
+        }
+      };
+      const validation = parameterValidation(
+        schema,
+        req
+      );
+
+      expect(validation)
+        .to.be.a('boolean');
+
+      expect(validation)
+        .to.be.true;
+
+      expect(req.params.id)
+        .to.be.a('Date');
+
+      expect(req.params.id.toString())
+        // I hate that month is 0-index
+        .to.equal(new Date(1943, 3, 16).toString());
+    });
+
+    // it('to a string', () => {
```

Comment:
Joi does not seem to want to convert to a string - will see what I can do about it.